usr/src/cmd/zfs/zfs_main.c

```diff
@@ -4440,7 +4444,7 @@ parse_fs_perm(fs_perm_t *fsperm, nvlist_t *nvl)
 		nvlist_t *nvl2 = NULL;
 		const char *name = nvpair_name(nvp);
 		uu_avl_t *avl = NULL;
-		uu_avl_pool_t *avl_pool;
+		uu_avl_pool_t *avl_pool = NULL;
```

Comment:
This is probably wrong (dereferences NULL instead of uninitialized memory), as is the use of "is_set" below, and the "default" case.  But I don't really care to review this thoroughly -- you should check with the original authors.